usb/esp_tinyusb/tusb_msc_storage_spiffs.c

```diff
@@ -0,0 +1,196 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// DESCRIPTION:
+// This file contains the code for accessing the storage medium i.e. SPI Flash.
+
+#include "esp_log.h"
+#include "esp_err.h"
+#include "esp_check.h"
+#include "esp_vfs_fat.h"
+#include "diskio_impl.h"
+#include "diskio_wl.h"
+#include "wear_levelling.h"
+#include "esp_partition.h"
+#include "tusb_msc_storage.h"
+
+static wl_handle_t s_wl_handle = WL_INVALID_HANDLE;
+static bool s_fat_mounted;
+static const char *s_base_path;
+
+static const char *TAG = "msc_spiflash";
+
+esp_err_t tinyusb_msc_storage_init(const tinyusb_msc_storage_config_t *config)
+{
+    assert(config->storage_type == TINYUSB_MSC_STORAGE_SPI);
+    ESP_LOGI(TAG, "Initializing wear levelling");
+
+    const esp_partition_t *data_partition = esp_partition_find_first(ESP_PARTITION_TYPE_DATA, ESP_PARTITION_SUBTYPE_DATA_FAT, NULL);
+    if (data_partition == NULL) {
+        ESP_LOGE(TAG, "Failed to find FATFS partition. Check the partition table.");
+        return ESP_ERR_NOT_FOUND;
+    }
+
+    return wl_mount(data_partition, &s_wl_handle);
+}
+
+static inline size_t esp_vfs_fat_get_allocation_unit_size(
+    size_t sector_size, size_t requested_size)
+{
+    size_t alloc_unit_size = requested_size;
+    const size_t max_sectors_per_cylinder = 128;
+    const size_t max_size = sector_size * max_sectors_per_cylinder;
+    alloc_unit_size = MAX(alloc_unit_size, sector_size);
+    alloc_unit_size = MIN(alloc_unit_size, max_size);
+    return alloc_unit_size;
+}
+
+esp_err_t tinyusb_msc_storage_mount(const char *base_path)
```

Comment:
Likewise, why is this necessary when there is already `esp_vfs_fat_spiflash_mount_rw_wl`?

Reply:
Above explanation of `esp_vfs_fat_sdmmc_mount ` holds good for `esp_vfs_fat_spiflash_mount_rw_wl ` as well.